CONTRIBUTING.md

```diff
@@ -0,0 +1,204 @@
+- organizing, branching, and merging this repository;
+- producing and publishing release artifacts; and
+- feature and release planning.
+
+This is a living document that is not (and possibly cannot be)
+comprehensive. If something is missing or unclear, or if you have
+suggestions for improving our processes, please file an issue or open a
+pull request.
+
+# Building
+
+See guidance in [README.md](README.md) for information on building SAW.
+
+# Testing
+
+The core test suite for SAW consists of a collection of integration
+tests, in subdirectories of the `intTest` directory. A Cabal test suite
+called `integration_tests` is responsible for invoking these tests.
+
+## Dependency freezing
+
+To ensure repeatable test results, our CI system uses
+`cabal.project.freeze` files to fix the version of all dependencies
+prior to building. It uses one of several version freezing files, each
+named `cabal.GHC-<VER>.config`, for GHC version `<VER>`.
+
+We recommend building and testing using these files unless your explicit
+goal is to assess compatibility with different library versions. To do
+so, run the following before building or testing:
+
+    ln -s `cabal.GHC-<VER>.config cabal.project.freeze
+
+## Running and Creating Tests
+
+For more information on running and creating tests, see the [README.md
+file in the intTests directory](intTests/README.md).
+
+## Other Tests in CI
+
+Although individual developers typically run just the core integration
+tests, our CI system (GitHub Actions) runs several other tests on every
+pull request. These include ensuring that a variety of large proof
+developments, many for the [s2n TLS
+library](https://github.com/aws/s2n-tls), continue to succeed. The
+scripts for running these tests are located in the [s2nTests](s2nTests)
+directory.
+
+# Repository organization and practices
+
+The top-level repository directories are:
+
+* `src` - Source code for the SAWScript interpreter library.
+
+* `saw` - Source for the `saw` executable interpreter for SAWScript.
+
+* `doc` - A tutorial and manual for SAWScript.
+
+* `examples` - Various examples of using SAWScript.
+
+* `intTests` - The central test suite for SAWScript.
+
+* `s2nTests` - Additional tests for SAWScript, confirming that
+  verifications involving the s2n library continue to work.
+
+* `saw-remote-api` - Source for a server that provides a remote API to
+  SAW based on JSON-RPC, and a Python client for that API.
+
+* `saw-core` - Source code for the SAWCore intermediate language used
+  within SAW.
+
+* `saw-core-aig` - A library for translating SAWCore to And-Inverter
+  Graphs.
+
+* `saw-core-coq` - A library for translating SAWCore to Coq's Gallina
+  language.
+
+* `saw-core-sbv` - A library for translating SAWCore predicates to SMT
+  queries using the [SBV](http://leventerkok.github.io/sbv/) library.
+
+* `saw-core-what4` - A library for translating SAWCore predicates to
+  solver queries using the [What4](https://github.com/galoisinc/what4)
```

Comment:
It doesn't matter to github, but we might want to use the normal GaloisInc capitalization method in the URL.

Reply:
Fixed.